File: core/Application.cpp
/*
 *  Application.cpp
 *  core
 *
 *  Created by Tymon Tobolski on 10-08-23.
 *  Copyright 2010 Politechnika Wrocławska. All rights reserved.
 *
 */

#include "Application.h"

Object* Application::createObject(TuioObject *tobj){
	return new Object(tobj);
}

Cursor* Application::createCursor(TuioCursor *tcur){
	return new Cursor(tcur);
}

void Application::display(){
	
}

File: src/main/scala/synergia/platri/Application.scala
package synergia.platri

import TUIO._

trait Application extends GFX {
    def createObject(tobj: TuioObject): Object = new Object(tobj)

    def createCursor(tcur: TuioCursor): Cursor = new Cursor(tcur)

    def display {}

    def start { Debug.info("Application START") }

    def stop { Debug.info("Application STOP") }

    protected def sym(tobj: TuioObject) = tobj.getSymbolID
}
